service/kinesis/customizations.go

```diff
@@ -0,0 +1,26 @@
+package kinesis
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+var readDuration = 15 * time.Second
+
+func init() {
+	ops := []string{
+		opGetRecords,
+	}
+	initRequest = func(r *request.Request) {
+		for _, operation := range ops {
+			if r.Operation.Name == operation {
+				r.Handlers.Unmarshal.PushFront(addTimeoutReadCloserOption)
```

Comment:
With the suggestion above of fixing, `WithResponseReadTimeout`  This line becomes the following, No need for the `addTimeoutReadCloserOption` function.

```go
r.ApplyOptions(request.WithResponseReadTimeout(readDuration))
```